Add package reference docs (#2046)
OctopusDeploy/docs: src/pages/docs/deployments/kubernetes/kustomize/index.md

The target file paths are relative to the root of the git repository.
Again remember that in Linux workers, paths are case-sensitive, so it is always good practice to check this.  
You can use glob patterns to target multiple files. [Learn more about glob patterns](/docs/deployments/kubernetes/glob-patterns).

## Referencing Container images

> Since v2.0.1, you are able to add container images as package references directly from the Kustomize step. Image references added in this way can be used via [variable substitutions](/docs/projects/variables/variable-substitutions).

Reviewer: ```suggestion
From v2.0.1, you are able to add container images as package references directly from the Kustomize step. Image references added in this way can be used via [reference package variables](/docs/projects/variables/system-variables#reference-package-variables).
```

:::figure
![Add package references](/docs/deployments/kubernetes/kustomize/referenced-packages.png)
:::

For example, if you add a container image reference for `nginx`:

:::figure
![Reference a package](/docs/deployments/kubernetes/kustomize/reference-a-package.png)
:::

> You will then be able to select the version of this container image at release creation time. You can use the referenced `nginx` container image in a YAML file as a variable:

Reviewer: ```suggestion
You will then be able to select the version of this container image at release creation time. You can use the referenced `nginx` container image in your `kustomization.yaml` file using the following syntax:
```

> ```yaml

Reviewer: ```yaml
# ~/myApp/kustomization.yaml
resources:
- deployment.yaml

images:
- name: nginx
   newTag: #{Octopus.Action.Package[nginx].PackageVersion}
```

Reviewer: I think this may be a better example, more the kustomize way 😉
